src/core/contracts/identity.ts

```diff
@@ -183,6 +183,185 @@ const getBeaconAddress = async (opts?: ConfigOpts): Promise<EthereumAddress> =>
+  primaryType: string;
+  domain: TypedDomainData;
+  message: Record<string, unknown>;
+}
+
+type TypedMessageDataField = Record<string, unknown>;
+
+/**
+ * EIP712Signature represents a ECDSA r value, s value and EIP-155 calculated Signature v value
+ */
+export type EIP712Signature = { r: string; s: string; v: number };
+
+/**
+ * TypedDataField represents a single type
+ */
+export interface TypedDataField {
+  name: string;
+  type: string;
+}
+
+const EIP712_DOMAIN_TYPES: TypedDataField[] = [
+  { name: "name", type: "string" },
+  { name: "version", type: "string" },
+  { name: "chainId", type: "uint256" },
+  { name: "verifyingContract", type: "address" },
+  { name: "salt", type: "bytes32" },
+];
+
+/**
+ * createTypeData() Allows users to create a JSON-Schema definition for EIP-712 TypedData params.
+ *
+ * @param domainData - Unique data that identifies a contract to prevent phishing attacks
+ * @param primaryType - A top-level type
+ * @param message - The message to be signed
+ * @param messageTypes - A set of all structured types for messages
+ * @returns Typed structured data to be signed
+ */
+export const createTypeData = (
+  domainData: TypedDomainData,
+  primaryType: string,
+  message: TypedMessageDataField,
+  messageTypes: Record<string, Array<TypedDataField>>
+): TypedData => {
+  return {
+    types: {
+      EIP712Domain: EIP712_DOMAIN_TYPES,
+      ...messageTypes,
+    },
+    domain: domainData,
+    primaryType,
+    message,
+  };
+};
+
+/**
+ * getDomainSeperator() Gets unique data that identifies the Identity contract to prevent phishing attacks
+ *
+ * @param contractAddress - Address of the identity contract to use
+ * @param opts - Optional. Configuration overrides, such as store, if any
+ * @returns A EIP-712 domain structure that is unique to the Identity contract
+ */
+export const getDomainSeperator = async (
+  contractAddress: EthereumAddress,
+  opts?: ConfigOpts
+): Promise<TypedDomainData> => {
+  const provider = requireGetProvider(opts);
+
+  return {
+    name: "Identity",
+    version: "1",
+    chainId: (await provider.getNetwork()).chainId,
+    verifyingContract: contractAddress,
+    salt: "0xa0bec69846cdcc8c1ba1eb93be1c5728385a9e26062a73e238b1beda189ac4c9",
+  };
+};
+
+/**
+ * createAddDelegateEip712TypedData() Allows users to create typed structure data for signing addDelegate messages
+ * following the EIP-712 standard.
+ *
+ * @param contractAddress - Address of the identity contract to use
+ * @param message - The DelegateAdd message to be signed
+ * @param opts - Optional. Configuration overrides, such as store, if any
+ * @returns Typed structure data for signing addDelegate messages
+ */
+export const createAddDelegateEip712TypedData = async (
+  contractAddress: EthereumAddress,
+  message: DelegateAddParams,
+  opts?: ConfigOpts
+): Promise<TypedData> => {
+  const primaryType = "DelegateAdd";
+  const TypedAddDelegate: TypedDataField[] = [
```

Comment:
`TypedAddDelegate` shouldn't be capitalized. Also `addDelegateType` makes more sense to me.